### PR TITLE
deps: Upgrade to RN v0.65!

### PR DESCRIPTION
**Pull request by chrisbobbe**
Fixes: #5230

### CODE REVIEW

**Comment by chrisbobbe** [comment]
Here are the RN commits that touch the [RN template app](https://github.com/facebook/react-native/tree/main/template) that are in v0.64.2 but are not in v0.65.2. They're ordered from first to last (oldest to newest).

I.e., they're the commits in the output of `git log --oneline --reverse v0.64.2..v0.65.2 -- template`, when I run it in a clone of the facebook/react-native repo.

Along with whether, how, and when we propagate the changes to our app.

We shouldn't forget to update this, if necessary, to follow any changes made during code review and merge.

| RN commit | When resolved | Comment |
| --- | --- | --- |
| d8b0997aa Upgrade metro to 0.64.0 | cabab2243 |  |
| 8f5656b34 Update to eslint@7.12 | b6546e9ae, 7e092787d |  |
| 58c80d4f8 Use codegen from source in default iOS template apps | N/A | We will end up adding the react-native-codegen dependency; see main commit. This RN commit was backported to RN v0.64 in facebook/react-native@68110d4ab and we processed it for that upgrade; see https://github.com/zulip/zulip-mobile/pull/4991#issuecomment-914724039 |
| f1f716879 Deploy Flow v0.138.0 | a5235e397 |  |
| 7f53bb95f Suppress errors and switch to new-generics in xplat | N/A | This gets cleanly reverted in facebook/react-native@b20698d81 in the template app. |
| e99b8bbb4 Use react-native-codegen@0.0.6 in new app template | N/A | We will end up adding the react-native-codegen dependency; see main commit. This RN commit was backported to RN v0.64 in facebook/react-native@65003d743 and we processed it for that upgrade; see https://github.com/zulip/zulip-mobile/pull/4991#issuecomment-914724039 |
| c901c1fbc Integrate Native Module codegen into Xcode build pipeline (#30449) | N/A | This was backported to RN v0.64 in facebook/react-native@d8a0a9b1d and we processed it for that upgrade; see https://github.com/zulip/zulip-mobile/pull/4991#issuecomment-914724039 |
| a326a30e3 Add instructions to template/ios/Podfile for enabling hermes (#30461) | 9a68ea3ea | This was backported to RN v0.64 in facebook/react-native@eb85d1dbd and we processed it for that upgrade; see https://github.com/zulip/zulip-mobile/pull/4991#issuecomment-914724039 |
| 652e3953f Update template devDependencies (#30489) | ce11ee6af | This was backported to RN v0.64 in facebook/react-native@fffa4d1ae and we processed it for that upgrade; see https://github.com/zulip/zulip-mobile/pull/4991#issuecomment-914724040 |
| 11cf9ca8c Deploy Flow v0.140.0 | a5235e397 |  |
| fdcacd7f7 fix: building in release mode for simulator (#30543) | 534be818c | This was backported to RN v0.64 in facebook/react-native@5c09b3fb5 and we processed it for that upgrade; see https://github.com/zulip/zulip-mobile/pull/4991#issuecomment-914724042 |
| e54ead655 fix: default template on iOS (#30571) | 9a68ea3ea | This was backported to RN v0.64 in facebook/react-native@2ba3ef5b1 and we processed it for that upgrade; see https://github.com/zulip/zulip-mobile/pull/4991#issuecomment-914724043 |
| 42dde12aa Exclude `i386` from valid architectures when building with Hermes on iOS (#30592) | 534be818c | This was backported to RN v0.64 in facebook/react-native@ed237b479 and we processed it for that upgrade; see https://github.com/zulip/zulip-mobile/pull/4991#issuecomment-914724044 |
| a4920ed90 Deploy 0.142.0 to xplat | main commit |  |
| a1d626739 Delete iOS10/tvOS10 support from React Native Libraries | c953bc336 | We already dropped iOS 10 support on our own schedule, not as part of a RN upgrade. |
| f2c6279ca iOS/tvOS 10.0 ==> 11.0 in .xcodeproj | c953bc337 | We already dropped iOS 10 support on our own schedule, not as part of a RN upgrade. |
| b20698d81 Remove generate-tests setting from flowconfigs in xplat | N/A | This cleanly reverts facebook/react-native@7f53bb95f in the template app; we didn't take that; ignore. |
| 8d81ae8f2 Deploy Flow v0.143.1 | main commit |  |
| cd6c9f327 Update react-native-codegen for template project (#30896) | main commit | We will end up adding the react-native-codegen dependency; see main commit. |
| e2263b343 Deploy Flow v0.144.0 | main commit |  |
| eae7e97f8 Bump Metro to v0.65.1 | 46f5264c5 |  |
| c7efd5b36 Bump Android compileSdkVersion and targetSdkVersion to 30 | 30b4d6a64 | We did this independently of any RN upgrade. |
| 2ecdb69b7 Revert D26445966: Bump Android compileSdkVersion and targetSdkVersion to 30 | N/A | We did this independently of any RN upgrade in 30b4d6a64, and we're not interested in reverting/reapplying it a bunch of times. |
| 9abafd1c2 Deploy v0.145.0 to xplat | main commit |  |
| 3399896ae Update flipper in RNTester and template (#31010) | N/A | This was backported to RN v0.64 in facebook/react-native@7159bcbf7 and we processed it for that upgrade; see https://github.com/zulip/zulip-mobile/pull/4991#issuecomment-914724042 |
| 55c883381 Bump Android compileSdkVersion and targetSdkVersion to 30 | N/A | We did this independently of any RN upgrade in 30b4d6a64, and we're not interested in reverting/reapplying it a bunch of times. |
| 519418303 Revert D26470604: Bump Android compileSdkVersion and targetSdkVersion to 30 | N/A | We did this independently of any RN upgrade in 30b4d6a64, and we're not interested in reverting/reapplying it a bunch of times. |
| 6ccd0cdeb Bump Android compileSdkVersion and targetSdkVersion to 30 (#31078) | N/A | We did this independently of any RN upgrade in 30b4d6a64, and we're not interested in reverting/reapplying it a bunch of times. |
| 84a81dac1 Upgrade Metro to 0.65.2 | 46f5264c5 |  |
| d477f8011 Deploy Flow v0.146.0 | main commit |  |
| 3f571203a Deploy Flow v0.147.0 | main commit |  |
| d04ec4dc6 Deploy v0.148.0 to xplat | main commit |  |
| 2785ce7e6 React Native sync for revisions 6d3ecb7...c9aab1c | main commit |  |
| 7258afeea bump gradle wrapper to 6.8.3 (#31347) | fb0cf3103 | We went to 6.9 during an Expo upgrade. |
| f1d1e23b2 bump react-native-github's metro version to v0.66.0 | 46f5264c5 |  |
| 06f52f766 Deploy v0.149.0 to xplat | main commit |  |
| 73844712b Fixing the git attrs for all the people and all the files and all future (#31128) | N/A | This was backported to RN v0.64 in facebook/react-native@728d55a3e and we processed it for that upgrade; see https://github.com/zulip/zulip-mobile/pull/4991#issuecomment-914724042 |
| 7599593b3 Gradle 6.9, Android Gradle Plugin 4.2.1 (#31593) | 53ed19500 |  |
| 9a923be89 remove jcenter (#31609) | 1f80f4b6c | Unfortunately we have to carve out a temporary exception for something that's still using jcenter. |
| 37e9f1d36 bump buildToolsVersion to 30.0.2 (#31627) | N/A | We don't set this explicitly; we leave it up to the Android plugin to apply to our app project. |
| 5f30232c4 [0.65.0-rc.0] Bump version numbers | N/A |  |
| 4b62cbdb4 Revert "[0.65.0-rc.0] Bump version numbers" | N/A |  |
| e32449894 [0.65.0-rc.0] Bump version numbers | N/A |  |
| 4e9ae4c98 Revert "[0.65.0-rc.0] Bump version numbers" | N/A |  |
| 555696864 (tag: v0.65.0-rc.0) [0.65.0-rc.0] Bump version numbers | N/A |  |
| a10a20105 Revert "Gradle 6.9, Android Gradle Plugin 4.2.1 (#31593)" | N/A | This reversion is reverted in facebook/react-native@be9a66999; ignore. |
| 0a15927dc Revert "bump buildToolsVersion to 30.0.2 (#31627)" | N/A | This reversion is reverted in facebook/react-native@ba4424fcc; ignore. |
| d8b115afa Bump Android deps to 0.93 (#31675) | 7b4be5d33 |  |
| 0e08b2528 (tag: v0.65.0-rc.1) [0.65.0-rc.1] Bump version numbers | N/A |  |
| be9a66999 Revert "Revert "Gradle 6.9, Android Gradle Plugin 4.2.1 (#31593)"" | N/A | This reverts the reversion in facebook/react-native@a10a20105, which we didn't take; ignore. |
| ba4424fcc Revert "Revert "bump buildToolsVersion to 30.0.2 (#31627)"" | N/A | This reverts the reversion in facebook/react-native@0a15927dc, which we didn't take; ignore. |
| db7aa7b12 (tag: v0.65.0-rc.2) [0.65.0-rc.2] Bump version numbers | N/A |  |
| e8d725a37 (tag: v0.65.0-rc.3) [0.65.0-rc.3] Bump version numbers | N/A |  |
| 98e173445 fix: Move react-native-codegen to be a direct dependency of react-native (fix for 0.65-stable) | N/A | This is reverted in facebook/react-native@e77595784; ignore. We will end up adding the react-native-codegen dependency; see main commit. |
| e77595784 Revert "fix: Move react-native-codegen to be a direct dependency of react-native (fix for 0.65-stable)" | N/A | This reverts facebook/react-native@98e173445, which we didn't take; ignore. We will end up adding the react-native-codegen dependency; see main commit. |
| 5f0b805b5 (tag: v0.65.0-rc.4) [0.65.0-rc.4] Bump version numbers | N/A |  |
| 7473ce1d4 (tag: v0.65.0) [0.65.0] Bump version numbers | N/A |  |
| 5a1dc1b6b Handle OSS renderers in sync script | main commit |  |
| 8b430e00d fix AGP 7 compatibility (#32030) | N/A | The changed code is relevant only for projects that use Buck. We don't, and we don't have this code; see our 1c86488ce. |
| e4d576f65 (tag: v0.65.1) [0.65.1] Bump version numbers | N/A |  |
| 7eff4b6f8 (tag: v0.65.2, origin/0.65-stable) [0.65.2] Bump version numbers | main commit |  |

File: package.json [hunk 1]
```diff
@@ -49,9 +49,10 @@
     "lodash.omit": "^4.5.0",
     "lodash.union": "^4.6.0",
     "lodash.uniqby": "^4.4.0",
-    "react": "17.0.1",
+    "react": "17.0.2",
     "react-intl": "5.24.6",
-    "react-native": "0.64.3",
+    "react-native": "0.65.2",
+    "react-native-codegen": "^0.0.7",
```
Comment (gnprice) [inline package.json:55]: It looks like in the template app, `react-native-codegen` goes in `devDependencies`. Can we similarly put it there?

**Review by gnprice — COMMENTED** [review]
Thanks @chrisbobbe for all your detailed work on this! Including that table of upstream commits, and the series of prep PRs that got our libdefs up to snuff.

Just one small comment below, and a followup opportunity I noticed that's actually about a previous upgrade. Otherwise all looks good!

**Comment by chrisbobbe** [comment]
Thanks for the review! Revision pushed.

**Comment by gnprice** [comment]
Thanks! Looks good; merging.

I've added a small docs commit on that profiling update, and also tweaked the last commit's summary-line prefix to be a bit more general (moving "checkCompatibility" into the rest of the summary line, and saying "api" instead which is broad enough to be something that recurs.)

**Comment by chrisbobbe** [comment]
Thanks!

**Pull request merged**